sympy/series/tests/test_order.py

```diff
@@ -423,6 +423,11 @@ def test_issue_9910():
     assert O(x*log(x) + sin(x), (x, oo)) == O(x*log(x), (x, oo))
 
 
+def test_issue_9917():
+    assert O(x*sin(x) + 1, (x, oo)).expr.cancel() == x*sin(x) + 1
```

Comment:
This test now outputs `x` on master, and the test below outputs `x**2` (which is correct as `x->oo`).